lib/datasets/dataset.rb

```diff
@@ -33,9 +33,21 @@ def cache_path
       @cache_path ||= CachePath.new(@metadata.id)
     end
 
-    def download(output_path, url, &block)
-      downloader = Downloader.new(url)
-      downloader.download(output_path, &block)
+    def download(output_path, *urls, &block)
+      urls.each do |url|
+        downloader = Downloader.new(url)
+        downloader.download(output_path, &block)
+        return
+      rescue Net::HTTPClientException => error
+        if urls.last != url
+          message = "site is not available: " +
+                    "#{error.class}: #{error.message}: " +
+                    "Attempting to download from an alternative site"
```

Comment:
We should include the failed URL and fallback URL to be tried for easy to debug.

Reply:
I agree, including the failed URL and fallback URL will make debugging easier. I would like to think a debug friendly message.

Reply:
I implemented in d5cf668

Reply:
Mixing `:` and `..` as separator isn't a good idea.
How about 'fallback: <http://yann.lecun.com/exdb/mnist/t10k-labels-idx1-ubyte.gz> -> <https://ossci-datasets.s3.amazonaws.com/mnist/t10k-labels-idx1-ubyte.gz>` or something?

Reply:
Thanks for your suggestion. `->` separator is good idea! and angle brackets is used to delimit the boundaries of the URL in <https://www.ietf.org/rfc/rfc1738.txt>. It's reasonable!

> For this purpose,
   is recommended that angle brackets ("<" and ">"), along with the
   prefix "URL:", be used to delimit the boundaries of the URL.

Reply:
I implemented in 01e273e. e.g. : `403 Forbidden: fallback: <http://yann.lecun.com/exdb/mnist/t10k-labels-idx1-ubyte.gz> -> <https://ossci-datasets.s3.amazonaws.com/mnist/t10k-labels-idx1-ubyte.gz>`